src/scanpy/preprocessing/_utils.py

```diff
@@ -33,7 +33,10 @@ def _(X: np.ndarray, *, axis: Literal[0, 1], dtype: DTypeLike) -> np.ndarray:
 def _get_mean_var(
     X: _SupportedArray, *, axis: Literal[0, 1] = 0
 ) -> tuple[NDArray[np.float64], NDArray[np.float64]]:
-    if isinstance(X, sparse.spmatrix):
+    if isinstance(X, np.ndarray):
```

Comment:
instead of adding a second code path that handles np.ndarray, you should replace the existing one above:

```py
@axis_mean.register(np.ndarray)
def _(X: np.ndarray, ...): ...
```